flink-core/src/main/java/org/apache/flink/util/IOUtils.java

```diff
@@ -143,6 +143,30 @@ public static void readFully(final InputStream in, final byte[] buf, int off, fi
 		}
 	}
 
+	/**
+	 * Similar to readFully(). Returns the total number of bytes read into the buffer.
+	 * @param in
+	 *        The InputStream to read from
+	 * @param buf
+	 *        The buffer to fill
+	 * @return
+	 *        The total number of bytes read into the buffer
+	 * @throws IOException
+	 *         If the first byte cannot be read for any reason other than end of file,
+	 *         or if the input stream has been closed, or if some other I/O error occurs.
+	 */
+	public static int readFully(final InputStream in, final byte[] buf) throws IOException {
```

Comment:
nit: I would like to rename this to `tryReadFully`, as it has different usage contracts compared to the usual `readFully` method we're familiar with. e.g. this method doesn't throw when EOF is reached.

Reply:
Thanks, make sense.